src/Microsoft.AspNet.Razor/CodeGenerators/CSharpTagHelperCodeRenderer.cs

```diff
@@ -69,13 +69,11 @@ public void RenderTagHelper(TagHelperChunk chunk)
 
             RenderTagHelpersCreation(chunk, tagHelperDescriptors);
 
-            // Determine what attributes exist in the element and divide them up.
-            var htmlAttributes = chunk.Attributes;
-            var attributeDescriptors = tagHelperDescriptors.SelectMany(descriptor => descriptor.Attributes).ToArray();
-            var unboundHtmlAttributes = htmlAttributes.Where(attribute => !attributeDescriptors.Any(
-                attributeDescriptor => attributeDescriptor.IsNameMatch(attribute.Key)));
+            // This ensures that all created TagHelpers have initialized dictionary bound properties which are used
+            // via TagHelper indexers.
+            RenderTagHelperPropertyVerification(chunk.Attributes, tagHelperDescriptors);
```

Comment:
Not sure I agree with the "as simple as possible" goal but, if so, instead merge `RenderTagHelperPropertyVerification` into `RenderTagHelpersCreation()`. No need of a separate loop through both the descriptors and the attributes. If part of `RenderAttributes()` would avoid iterating through the attributes again; if part of `RenderTagHelpersCreation()`, save a loop through the descriptors.